runtime/connectors/s3/s3.go

```diff
@@ -192,3 +181,34 @@ func getAwsSessionConfig(ctx context.Context, conf *Config, bucket string) (*ses
 
 	return sess, nil
 }
+
+func resolvedCredentials(env *connectors.Env) (*credentials.Credentials, error) {
+	providers := make([]credentials.Provider, 0)
+	useHostCred := env.Variables["use_host_credentials"] != "false" // true by default
+	if useHostCred {
+		// The chain used here is a duplicate of defaults.CredProviders(), but without the remote credentials lookup (since they resolve too slowly).
+		providers = append(providers, &credentials.EnvProvider{}, &credentials.SharedCredentialsProvider{Filename: "", Profile: ""})
```

Comment:
Will need re-ordering given earlier comment about allow vs. use (can just add these after staticProvider in chain if allow)